compiler/luci/pass/src/FuseBCQPass.cpp

```diff
@@ -267,13 +267,124 @@ template <> class BCQFuser<1>
 
   bool is_bcqinfo_valid()
   {
-    // do_w_x should be int32 or bool type
+    std::set<int32_t> ref_prefix_set;
+
     for (auto n : _do_w_x)
     {
-      if (n.second->dtype() != loco::DataType::BOOL && n.second->dtype() != loco::DataType::S32)
+      // do_w_x should be BOOL type
+      if (n.second->dtype() != loco::DataType::BOOL)
+      {
+        LOGGER(l);
+        WARN(l) << "FuseBCQPass : do_w_x has wrong type" << std::endl;
+        return false;
+      }
+
+      // Generate reference prefix set
+      ref_prefix_set.insert(n.first);
+    }
+
+    for (auto n : _alpha)
+    {
+      // alpha should be FLOAT32 type
+      if (n.second->dtype() != loco::DataType::FLOAT32)
+      {
+        LOGGER(l);
+        WARN(l) << "FuseBCQPass : alpha has wrong type" << std::endl;
+        return false;
+      }
+
+      if (ref_prefix_set.find(n.first) == ref_prefix_set.end())
+      {
+        LOGGER(l);
+        WARN(l) << "FuseBCQPass : Invalid prefix is found" << std::endl;
```

Comment:
Q) from the condition, is the meaning like `Required prefix not found. Invalid file` ?

Reply:
At first, I wanted to check that `prefix` is valid when we access to `map`.
However, it looks weird to add the assertion in this function.
I will remove this check in this PR and create another PR for it :)